rules/solid/src/Rector/If_/ChangeAndIfToEarlyReturnRector.php

```diff
@@ -198,13 +201,17 @@ private function getBooleanAndConditions(BooleanAnd $booleanAnd): array
      * @param Expr[] $conditions
      * @return If_[]
      */
-    private function createInvertedIfNodesFromConditions(array $conditions): array
+    private function createInvertedIfNodesFromConditions(bool $isIfInLoop, array $conditions): array
     {
         $ifs = [];
         foreach ($conditions as $condition) {
             $invertedCondition = $this->conditionInverter->createInvertedCondition($condition);
             $if = new If_($invertedCondition);
-            $if->stmts = [new Return_()];
+            if ($isIfInLoop) {
```

Comment:
I'm looking int to this and the only difference is extra `return` after the `if`.

<br>

So basically here must be check for the `return;` in $foreach->stmts after the `if`.

Something like:

```suggestion
            if ($isIfInLoop || $this->isFollowedByReturn((array) $foreach->stmts) {
```